### ISSUE TITLE
terraform-fmt fails on valid syntax

### ISSUE THREAD

**Issue by aharden** [hashicorp/terraform-github-actions#28]
# Release
0.3.1

# Description
The `fmt` action fails on valid Terraform syntax that passes `terraform fmt` locally.

# Output
```
terraform fmt Failed
modules/roles/main.tf
 }
 
 resource "aws_iam_policy" "ec2-autoscaling-setdesiredcapacity" {
-  name        = "Ev-EC2-ASG-SetDesiredCapacity"
-  path        = "/"
+  name = "Ev-EC2-ASG-SetDesiredCapacity"
+  path = "/"
   description = "Provides ability to set desired capacity on ASGs."
 
   policy = <<EOF
```

#GitHub Action Workflow
```
workflow "Terraform" {
  resolves = "terraform-plan"
  on = "pull_request"
}

action "filter-to-pr-open-synced" {
  uses = "actions/bin/filter@master"
  args = "action 'opened|synchronize'"
}

action "terraform-fmt" {
  uses = "hashicorp/terraform-github-actions/fmt@v0.3.1"
  needs = "filter-to-pr-open-synced"
  secrets = ["GITHUB_TOKEN"]
  env = {
    TF_ACTION_WORKING_DIR = "."
  }
}

action "terraform-init" {
  uses = "hashicorp/terraform-github-actions/init@v0.3.1"
  needs = "terraform-fmt"
  secrets = [
    "GITHUB_TOKEN",
    "AWS_ACCESS_KEY_ID",
    "AWS_SECRET_ACCESS_KEY",
  ]
  env = {
    TF_ACTION_WORKING_DIR = "."
  }
}

action "terraform-validate" {
  uses = "hashicorp/terraform-github-actions/validate@v0.3.1"
  needs = "terraform-init"
  secrets = [
    "GITHUB_TOKEN",
    "AWS_ACCESS_KEY_ID",
    "AWS_SECRET_ACCESS_KEY",
    "DYN_PASSWORD",
  ]
  env = {
    TF_ACTION_WORKING_DIR = "."
  }
}

action "terraform-plan" {
  uses = "hashicorp/terraform-github-actions/plan@v0.3.1"
  needs = "terraform-validate"
  secrets = [
    "GITHUB_TOKEN",
    "AWS_ACCESS_KEY_ID",
    "AWS_SECRET_ACCESS_KEY",
    "DYN_PASSWORD",
  ]
  env = {
    TF_ACTION_WORKING_DIR = "."
    # If you're using Terraform workspaces, set this to the workspace name.
    TF_ACTION_WORKSPACE = "default"
  }
}
```

**Comment by lkysow** [comment]
Hi Alex, which version of Terraform are you running locally? These actions use 0.12.

**Comment by aharden** [comment]
Hi @lkysow, understood.  I'm running a mix of TF 0.11 and 0.12.  I explicitly ran TF 0.12.0's fmt before filing this issue.  This is happening on the first repo/config I'm running with TF 0.12.  I'm using v0.1.3 on my TF 0.11 repos without seeing an issue like this.

**Comment by lkysow** [comment]
Can you provide me a terraform file to reproduce?

**Comment by aharden** [comment]
Here is the `main.tf` that I observed the issue on - it wanted to "correct" lines 157 and 158.  Looking at the whole file, I'm wondering if the heredoc policies are throwing off `terraform fmt`:
```
locals {
  workspace = "${terraform.workspace}"
}

# Policy docs
data "aws_iam_policy_document" "instance-assume-role-policy" {
  statement {
    actions = ["sts:AssumeRole"]

    principals {
      type        = "Service"
      identifiers = ["ec2.amazonaws.com"]
    }
  }
}

data "aws_iam_policy_document" "ec2-rs-instance-assume-role-policy" {
  statement {
    actions = ["sts:AssumeRole"]

    principals {
      type = "Service"

      identifiers = [
        "ec2.amazonaws.com",
        "redshift.amazonaws.com",
      ]
    }
  }
}

data "aws_iam_policy_document" "rs-instance-assume-role-policy" {
  statement {
    actions = ["sts:AssumeRole"]

    principals {
      type        = "Service"
      identifiers = ["redshift.amazonaws.com"]
    }
  }
}

data "aws_iam_policy_document" "elastic_ip_associate" {
  statement {
    sid = "1"

    effect = "Allow"

    actions = [
      "ec2:AssociateAddress",
    ]

    resources = [
      "*",
    ]
  }
}

resource "aws_iam_policy" "athena-fullaccess" {
  name        = "Ev-AmazonAthenaFullAccess"
  path        = "/"
  description = "Allows full access to Athena without allowing list access to all S3 buckets."

  policy = <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "athena:*"
            ],
            "Resource": [
                "*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "glue:CreateDatabase",
                "glue:DeleteDatabase",
                "glue:GetDatabase",
                "glue:GetDatabases",
                "glue:UpdateDatabase",
                "glue:CreateTable",
                "glue:DeleteTable",
                "glue:BatchDeleteTable",
                "glue:UpdateTable",
                "glue:GetTable",
                "glue:GetTables",
                "glue:BatchCreatePartition",
                "glue:CreatePartition",
                "glue:DeletePartition",
                "glue:BatchDeletePartition",
                "glue:UpdatePartition",
                "glue:GetPartition",
                "glue:GetPartitions",
                "glue:BatchGetPartition"
            ],
            "Resource": [
                "*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket",
                "s3:ListBucketMultipartUploads",
                "s3:ListMultipartUploadParts",
                "s3:AbortMultipartUpload",
                "s3:CreateBucket",
                "s3:PutObject"
            ],
            "Resource": [
                "arn:aws:s3:::aws-athena-query-results-*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetObject",
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::athena-examples*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "sns:ListTopics",
                "sns:GetTopicAttributes"
            ],
            "Resource": [
                "*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "cloudwatch:PutMetricAlarm",
                "cloudwatch:DescribeAlarms",
                "cloudwatch:DeleteAlarms"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}
EOF
}

resource "aws_iam_policy" "ec2-autoscaling-setdesiredcapacity" {
  name        = "Ev-EC2-ASG-SetDesiredCapacity"
  path        = "/"
  description = "Provides ability to set desired capacity on ASGs."

  policy = <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "1",
            "Effect": "Allow",
            "Action": [
                "autoscaling:SetDesiredCapacity",
                "autoscaling:DescribeAutoScalingGroups"
            ],
            "Resource": "*"
        }
    ]
}
EOF
}
```

**Comment by lkysow** [comment]
When I run this locally I get the format behaviour you couldn't reproduce:
```bash
$ terraform fmt -diff -write=false main.tf
main.tf
--- old/main.tf
+++ new/main.tf
@@ -154,8 +154,8 @@
 }

 resource "aws_iam_policy" "ec2-autoscaling-setdesiredcapacity" {
-  name        = "Ev-EC2-ASG-SetDesiredCapacity"
-  path        = "/"
+  name = "Ev-EC2-ASG-SetDesiredCapacity"
+  path = "/"
   description = "Provides ability to set desired capacity on ASGs."

   policy = <<EOF
```

```bash
$ terraform version
Terraform v0.12.0

Your version of Terraform is out of date! The latest version
is 0.12.1. You can update by downloading from www.terraform.io/downloads.html
```

**Comment by aharden** [comment]
Hmm, I just ran `terraform-0.12.0 fmt modules/roles/main.tf` and it performed the correction that it didn't when I just ran `terraform-0.12.0 fmt` in the base directory of the configuration.  I wonder if I'm getting weird `fmt` behavior because I have TF 0.11.11 named `terraform` and TF 0.12.0 named `terraform-0.12.0`.  In any case, this is an oddity with TF 0.12, not the Github action, so I'll close the issue.  Thanks @lkysow!
